src/Core/Domain/MailTemplate/CommandHandler/GenerateThemeMailTemplatesCommandHandler.php

```diff
@@ -108,25 +108,21 @@ public function handle(GenerateThemeMailTemplatesCommand $command)
     }
 
     /**
-     * When installing a new Language, if it's a new one the Translator component can't manage it because its cache is
-     * already filled with the default one as fallback. We force the component to update its cache by adding a fake
-     * resource for this locale (this is the only way clean its local cache)
+     * When installing a new language, we must add the lang pack to the translator
      *
-     * @param string $locale
+     * @param $locale
      */
-    private function cleanTranslatorLocaleCache($locale)
+    private function addLanguageToTranslator($locale)
     {
-        if (!method_exists($this->translator, 'addLoader')
```

Comment:
Be careful, I remember this check was here for a reason because sometimes the injected translator was not `PrestaShopBundle\Translation\Translator` but `PrestaShopBundle\Translation\DataCollectorTranslator` which doesn't have the method `addRessource` nor `addLoader`
Although since there have been modifications on the translator (we had 2 different components) by @matthieu-rolland maybe this fixed the issue

Reply:
I think it's fixed but it should be tested.

FYI you get DataCollectorTranslator when you're in dev mode (it tracks translation hits for the Symfony profiler)

Reply:
Yeah, actually if I check if those methods exists it will return false because I got a `DataCollectorTranslator` **but** as `DataCollectorTranslator` is wrapping the `Translator` class, those methods are still available via the magic function `__call` that call the translator functions and make it works.